pkg/deploy/controller/deploymentconfig/controller.go

```diff
@@ -129,14 +139,21 @@ func (c *DeploymentConfigController) Handle(config *deployapi.DeploymentConfig)
 	if err != nil {
 		// If the deployment was already created, just move on. The cache could be
 		// stale, or another process could have already handled this update.
-		if errors.IsAlreadyExists(err) {
+		if kapierrors.IsAlreadyExists(err) {
 			return nil
 		}
 		c.recorder.Eventf(config, kapi.EventTypeWarning, "DeploymentCreationFailed", "Couldn't deploy version %d: %s", config.Status.LatestVersion, err)
 		return fmt.Errorf("couldn't create deployment for deployment config %s: %v", deployutil.LabelForDeploymentConfig(config), err)
 	}
 	c.recorder.Eventf(config, kapi.EventTypeNormal, "DeploymentCreated", "Created new deployment %q for version %d", created.Name, config.Status.LatestVersion)
 
+	// As we've just created a new deployment, we need to make sure to clean up old
+	// ones if we have reached our deployment history quota
+	existingDeployments.Items = append(existingDeployments.Items, created)
```

Comment:
Note that the only reason you need to append here is in case a user has specified `d.Spec.RHL: 0` and the deployment finished faster than the cleaning up below (which is almost(?) impossible)